Library/Formula/bison.rb

```diff
@@ -14,6 +14,8 @@ class Bison < Formula
 
   keg_only :provided_by_osx, "Some formulae require a newer version of bison."
 
+  depends_on "homebrew/dupes/m4" => :build if OS.linux?
```

Comment:
Build dependencies that are provided by Mac OS are `unless OS.mac?`. Same for disabling code and patches that are specific to Mac OS. Adding code that fixes a bug that's unique to Linux but doesn't apply to Mac OS would use `if OS.linux?`.